src/nvim/ops.c

```diff
@@ -2497,9 +2499,7 @@ static void op_yank_reg(oparg_T *oap, bool message, yankreg_T *reg, bool append)
    */
   curbuf->b_op_start = oap->start;
   curbuf->b_op_end = oap->end;
-  if (yanktype == MLINE
-      && !oap->block_mode
-      ) {
+  if (yanktype == MLINE) {
```

Comment:
The old code implies that `oap->block_mode` could be true even if `oap->motion_type == MLINE`. With this PR, it's not possible to have `MLINE` motion type and also a block-mode operation. That makes sense to me, but I wonder if I'm missing something.